lib/mbta_v3_api/store/predictions.ex

```diff
@@ -5,28 +5,153 @@ defmodule MBTAV3API.Store.Predictions do
 
   Based on https://github.com/mbta/dotcom/blob/main/lib/predictions/store.ex
   """
+  use GenServer
+  require Logger
+  alias MBTAV3API.Prediction
+
   @behaviour MBTAV3API.Store
 
+  @predictions_table_name :predictions_from_streams
+
+  @spec start_link(Keyword.t()) :: GenServer.on_start()
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  @impl true
+  def init(_) do
+    _table = :ets.new(@predictions_table_name, [:named_table])
+
+    {:ok, %{}}
+  end
+
+  @impl true
+  def fetch(fetch_keys) do
+    match_spec = prediction_match_spec(fetch_keys)
+
+    timed_fetch([{match_spec, [], [:"$1"]}], "fetch_keys=#{inspect(fetch_keys)}")
+  end
+
+  @impl true
+  def fetch_multi_filter(fetch_keys_list) do
+    match_specs =
+      fetch_keys_list
+      |> Enum.map(&prediction_match_spec(&1))
+      |> Enum.map(&{&1, [], [:"$1"]})
+
+    timed_fetch(match_specs, "multi_fetch=true fetch_keys=#{inspect(fetch_keys_list)}")
+  end
+
+  defp prediction_match_spec(fetch_keys) do
+    # https://www.erlang.org/doc/apps/erts/match_spec.html
+    # Match the fields specified in the fetch_keys and return the full prediction
+    # see to_record/1 for the defined order of fields
+    {
+      Keyword.get(fetch_keys, :prediction_id, :_) || :_,
+      Keyword.get(fetch_keys, :route_id, :_) || :_,
+      Keyword.get(fetch_keys, :stop_id, :_) || :_,
+      Keyword.get(fetch_keys, :direction_id, :_) || :_,
+      Keyword.get(fetch_keys, :trip_id, :_) || :_,
+      Keyword.get(fetch_keys, :vehicle_id, :_) || :_,
```

Comment:
I feel like if we're using `|| :_` to handle when `fetch_keys` has `vehicle_id: nil` we could just use `Keyword.get/2` so that we don't have to specify the fallback twice.

Reply:
`Keyword.get/3` will only use its fallback if the requested key is missing from the keyword list; if the key is present but `nil` then it will return `nil`. A hypothetical use case in which I can imagine this mattering:
```elixir
stop_id = Map.fetch!(params, "stop_id")
direction_id =
    if direction_id = Map.get(params, "direction_id") then
        String.to_integer(direction_id)
    end
MBTAV3API.Store.Predictions.fetch(stop_id: stop_id, direction_id: direction_id)
```
If there is no `direction_id` in `params`, then I think this will filter to predictions with a `direction_id` of `nil`. Using `Keyword.get(fetch_keys, :stop_id) || :_` would mean that `stop_id` being `nil` and `stop_id` being absent would be treated the same way, which I think is less likely to cause really confusing bugs in six months.

Reply:
Ah thanks, should have read your initial suggestion more carefully.